pkg/apis/pipeline/v1beta1/task_types.go

```diff
@@ -126,7 +126,16 @@ type Step struct {
 }
 
 // A sidecar has the same data structure as a Step, consisting of a Container, and optional Script.
-type Sidecar = Step
+type Sidecar struct {
+	corev1.Container `json:",inline"`
+
+	// Script is the contents of an executable file to execute.
+	// If Script is not empty, the Step cannot have an Command or Args.
+	Script string `json:"script,omitempty"`
+	// ForceTerminationDisabled is a bool flag to toggle sidecar termination when all the steps in the task are completed.
+	// +optional
+	ForceTerminationDisabled bool `json:"forceTerminationDisabled,omitempty,default:false"`
```

Comment:
Does the `default:false` here do anything? AFAIK the `json:""` struct tag only really supports things like `omitempty`, `inline`, etc.

Reply:
Worth testing. I thought it does 😄 
I think in this case `omitempty` makes sense then

Reply:
It really doesn't work!
And I learned the hard way that there's no native good way to set a boolean flag in go struct to "true" by default 🤦‍♂️ 

Thanks for the catch!